pkg/kubelet/volume/volume_manager.go

```diff
@@ -0,0 +1,357 @@
+
+	<-stopCh
+	glog.Infof("Shutting down Kubelet Volume Manager")
+}
+
+func (vm *volumeManager) GetVolumesForPod(
+	podName types.UniquePodName) container.VolumeMap {
+	return vm.getVolumesForPodHelper(podName, nil /* pod */)
+}
+
+func (vm *volumeManager) GetVolumesForPodAndAppendSupplementalGroups(
+	pod *api.Pod) container.VolumeMap {
+	return vm.getVolumesForPodHelper("" /* podName */, pod)
+}
+
+func (vm *volumeManager) GetVolumesInUse() []api.UniqueVolumeName {
+	attachedVolumes := vm.actualStateOfWorld.GetAttachedVolumes()
+	volumesInUse :=
+		make([]api.UniqueVolumeName, 0 /* len */, len(attachedVolumes) /* cap */)
+	for _, attachedVolume := range attachedVolumes {
+		if attachedVolume.PluginIsAttachable {
+			volumesInUse = append(volumesInUse, attachedVolume.VolumeName)
+		}
+	}
+
+	return volumesInUse
+}
+
+// getVolumesForPodHelper is a helper method implments the common logic for
+// the GetVolumesForPod methods.
+func (vm *volumeManager) getVolumesForPodHelper(
+	podName types.UniquePodName, pod *api.Pod) container.VolumeMap {
+	if pod != nil {
+		podName = volumehelper.GetUniquePodName(pod)
+	}
+	podVolumes := make(container.VolumeMap)
+	for _, mountedVolume := range vm.actualStateOfWorld.GetMountedVolumesForPod(podName) {
+		podVolumes[mountedVolume.OuterVolumeSpecName] =
+			container.VolumeInfo{Mounter: mountedVolume.Mounter}
+		if pod != nil {
+			err := applyPersistentVolumeAnnotations(
+				mountedVolume.VolumeGidValue, pod)
+			if err != nil {
+				glog.Errorf("applyPersistentVolumeAnnotations failed for pod %q volume %q with: %v",
+					podName,
+					mountedVolume.VolumeName,
+					err)
+			}
+		}
+	}
+	return podVolumes
+}
+
+func (vm *volumeManager) WaitForAttachAndMount(pod *api.Pod) error {
+	expectedVolumes := getExpectedVolumes(pod)
+	if len(expectedVolumes) == 0 {
+		// No volumes to verify
+		return nil
+	}
+
+	glog.Infof("Waiting for volumes to attach and mount for pod %q", format.Pod(pod)) // TODO drop to V(3)?.
+	uniquePodName := volumehelper.GetUniquePodName(pod)
+
+	// If pod is already mounted, force remount to reflect any changes in the
+	// referencing pod. This will only remouunt volumes whose plugins indicates
+	// they require remounting on pod updates (Atomically updating volumes,
+	// like Downward API, depend on this to update the contents of the volume).
+	vm.desiredStateOfWorldPopulator.ReprocessPod(uniquePodName)
+	vm.actualStateOfWorld.MarkRemountRequired(uniquePodName)
+
+	err := wait.Poll(
+		podAttachAndMountRetryInterval,
+		podAttachAndMountTimeout,
+		vm.verifyVolumesMountedFunc(uniquePodName, expectedVolumes))
+
+	if err != nil {
+		// Timeout expired
+		ummountedVolumes :=
+			vm.getUnmountedVolumes(uniquePodName, expectedVolumes)
+		if len(ummountedVolumes) == 0 {
+			return nil
+		}
+
+		return fmt.Errorf(
+			"timeout expired waiting for volumes to attach/mount for pod %q/%q. list of unattached/unmounted volumes=%v",
+			pod.Name,
+			pod.Namespace,
+			ummountedVolumes)
+	}
```

Comment:
Add an else here with a high V level to say that the mounts for this pod are complete